src/react/RegisterDevicePage.js

```diff
@@ -141,50 +146,75 @@ export default class RegisterDevicePage extends React.Component {
   }
 
   render() {
+    const {
+      deviceId,
+      isValidDeviceId,
+      isInvalidDeviceId,
+      sendIntrospection,
+      introspectionInterfaces
+    } = this.state;
+
     return (
       <SingleCardPage title="Register Device" backLink="/devices">
         <Form onSubmit={this.registerDevice}>
-          <Form.Row>
+          <Form.Row className="mb-2">
             <Form.Group as={Col} controlId="deviceIdInput">
               <Form.Label>Device ID</Form.Label>
               <Form.Control
                 type="text"
                 className="text-monospace"
                 placeholder="Your device ID"
-                value={this.state.deviceId}
+                value={deviceId}
                 onChange={this.updateDeviceId}
                 autoComplete="off"
                 required
-                isValid={this.state.isValidDeviceId}
-                isInvalid={this.state.isInvalidDeviceId}
+                isValid={isValidDeviceId}
+                isInvalid={isInvalidDeviceId}
```

Comment:
Why defining two variables and having two sources of truth when each variable can be derived from the other?

Reply:
Fair point, but out of scope for this PR. Will follow up with a new PR for this